contracts/JBveBanny.sol

```diff
@@ -284,46 +293,106 @@ contract JBveBanny is ERC721Votes, ERC721Enumerable, Ownable, ReentrancyGuard, J
-
-    // The new lock must be greater than the current lock.
-    if (_lockedUntil > _updatedLockedUntil) revert INVALID_LOCK_EXTENSION();
+  function extendLock(JBLockExtensionData[] calldata _lockExtensionData) external nonReentrant {
+    for (uint256 _i; _i < _lockExtensionData; _i++) {
+      // Get a reference to the extension being iterated.
+      JBLockExtensionData _data = _lockExtensionData[_i];
+
+      // Duration must match.
+      if (!_isLockDurationAcceptable(_data.updatedDuration))
+        revert JBErrors.INVALID_LOCK_DURATION();
+
+      // Get the specs for the token ID.
+      (
+        uint256 _count,
+        ,
+        uint256 _lockedUntil,
+        bool _useJbToken,
+        bool _allowPublicExtension
+      ) = getSpecs(_data.tokenId);
+
+      // If the operation isn't allowed publicly, check if the msg.sender is either the position owner or is an operator.
+      if (!_allowPublicExtension)
+        _requirePermission(ownerOf(_data.tokenId), projectId, JBStakingOperations.EXTEND_LOCK);
+
+      // No time remaining if the lock has expired.
+      uint256 _timeRemaining = (block.timestamp >= _lockedUntil)
+        ? 0
+        : _lockedUntil - block.timestamp;
+
+      // Calculate the updated time when this lock will end (in seconds).
+      uint256 _updatedLockedUntil = block.timestamp + _data.updatedDuration - _timeRemaining;
+
+      // The new lock must be greater than the current lock.
+      if (_lockedUntil > _updatedLockedUntil) revert INVALID_LOCK_EXTENSION();
+
+      // fetch the stored packed value.
+      uint256 packedValue = _count;
+      // _duration in the bits 152-199.
+      packedValue |= _data.updatedDuration << 152;
+      // _lockedUntil in the bits 200-247.
+      packedValue |= _updatedLockedUntil << 200;
+      // _useJbToken in bit 248.
+      if (_useJbToken) packedValue |= 1 << 248;
+      // _allowPublicExtension in bit 249.
+      if (_allowPublicExtension) packedValue |= 1 << 249;
+
+      _packedSpecs[_data.tokenId] = packedValue;
+
+      emit ExtendLock(_data.tokenId, _data.updatedDuration, _updatedLockedUntil, msg.sender);
+    }
+  }
 
-    // fetch the stored packed value.
-    uint256 packedValue = _count;
-    // _duration in the bits 152-199.
-    packedValue |= _updatedDuration << 152;
-    // _lockedUntil in the bits 200-247.
-    packedValue |= _updatedLockedUntil << 200;
-    // _useJbToken in bit 248.
-    if (_useJbToken) packedValue |= 1 << 248;
+  /**
+    @notice
+    Allows banny holders to set whether or not anyone in the public can extend their locked position.
 
-    _packedSpecs[_tokenId] = packedValue;
+    @dev
+    Only an owner account or a designated operator can extend the lock of its tokens.
 
-    emit ExtendLock(_tokenId, _updatedDuration, _updatedLockedUntil, msg.sender);
+    @param _allowPublicExtensionData An array of locks to extend.
+  */
+  function setAllowPublicExtensionFlag(
```

Comment:
@mejango what was the reason for having this method?
Just curious, Have the DAO members (JBX Holders) requested this?

Reply:
yeah it was a response to concerns that people would have to constantly pay gas to get their full voting weight. Instead, they can allow anyone to pay the gas on their behalf.